packages/react/src/Button/Button.features.stories.tsx

```diff
@@ -16,44 +18,81 @@ export const LeadingVisual = () => <Button leadingVisual={HeartIcon}>Leading vis
 
 export const TrailingVisual = () => <Button trailingVisual={EyeIcon}>Trailing visual</Button>
 
+const AccessibilityNote = () => {
+  {
+    return (
+      <>
+        <p>
+          <b>Accessibility note</b>: If a button is dynamically updated to communicate a change (e.g. an action was
+          successful), please make sure that this is also properly communicated to screen reader users. This may not
+          happen reliably without additional markup considerations. Make sure to choose an approach that is appropriate
+          for your usecase.
+        </p>
+        <p>
+          Learn more about at{' '}
+          <Link href="https://github.com/github/accessibility/blob/8b300b36d8bca28fd5e3e70ffa077a6f8ee65c05/docs/wiki/screen-reader-testing/dynamically-updated-buttons-support-april-2024.md">
+            Staff-only: Dynamically updated button labels
+          </Link>
+          .
+        </p>
+      </>
+    )
+  }
+}
 export const TrailingCounter = () => {
   const [count, setCount] = useState(0)
+  const onClick = () => {
+    setCount(count + 1)
+    announce(`Watch ${count + 1}`)
```

Comment:
Hey @joshblack! I decided to update the approaches in these examples to use a live region, based on the [conclusion](https://github.com/github/accessibility/blob/main/docs/wiki/screen-reader-testing/dynamically-updated-buttons-support-april-2024.md#takeaways) I came to that the `aria-live` hack shouldn't be used, unless it's already being considered for other reasons.

I am thinking the next steps for me after this PR are:
* a dedicated storybook page that dives into different approaches for communicating button state.
* adding a section to the interface guidelines.